.github/workflows/build.yml

```diff
@@ -42,30 +42,30 @@ jobs:
         run: echo "Triggered by ${{ github.event_name }}"
 
       - name: Checkout code
-        uses: actions/checkout@v4
+        uses: actions/checkout@v4.3.0
```

Comment:
Not a maintainer but I suggest using hashes if you want true immutability. Anyone can overwrite that release: https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash

Reply:
My understanding is that releases used to be over-writable, but are now immutable if the [action's maintainer opts in](https://docs.github.com/en/actions/how-tos/create-and-publish-actions/using-immutable-releases-and-tags-to-manage-your-actions-releases). Short of 2 actions most have opted in, including [the one above](https://github.com/actions/checkout/pkgs/container/checkout).
<img width="532" height="94" alt="Screenshot 2025-09-18 at 5 54 25 PM" src="https://github.com/user-attachments/assets/96be3781-637e-4952-b9c4-c47df90d3b5b" />

Reply:
Ah, this is new to me. It is public preview atm.